crates/next-core/src/page_source.rs

```diff
@@ -301,11 +306,11 @@ async fn create_page_source_for_file(
             runtime_entries,
         )
     } else {
-        let data_pathname = StringVc::cell(format!(
-            "_next/data/development/{}",
-            get_asset_path_from_route(&pathname.await?, ".json")
-        ));
-        let data_path_regex = regular_expression_for_path(data_pathname);
+        let data_params_matcher = NextParamsMatcherVc::new(
+            pathname,
+            OptionStringVc::cell(Some("_next/data/development/".to_string())),
+            OptionStringVc::cell(Some(".json".to_string())),
+        );
```

Comment:
This is part 1 of the fix: ensuring that the params matcher ignores both the prefix and suffix of data requests.